### ISSUE TITLE
Feature Request: Resource upload dialog

### ISSUE THREAD

**Issue by gschueler** [rundeck/rundeck#134]
Would like to have a resource upload dialog which are attached to Jobs and can be defined as parameters for steps in the workflow execution.

Example: Uploading a tar-ball/ZIP which is used as input for a shell-script which untars the file on a host and runs a script within.

Tags: UX, Options, CustomerRequest

[original LH ticket](http://rundeck.lighthouseapp.com/projects/59277/tickets/214)

Original Creator: John Wyles

**Comment by walterhan** [comment]
+1 any discussion/updates on this?

**Comment by chrisking** [comment]
+1 This would be incredibly useful.

**Comment by navday** [comment]
+1 This would be extremely useful

**Comment by gentunian** [comment]
I agree it would be very helpful. Today we have a JOB that processes files. Those files are uploaded using FTP. Believe or not, some analyst asked me "how to use FTP". Despite that example, I think it would make rundeck more lovely! :)

**Comment by karouf** [comment]
I'm also considering Rundeck to build a self-service interface for non-tech people and one of the most frequent tasks they ask us to do is upload files on their behalf. I was thinking of using some FTP space dedicated to each project like @gentunian mentionned but it would be so much easier to benefit from Rundeck ACLs. I'm not a Java guy so can't really help with code but if I can do anything else to get this feature into Rundeck I'd be glad to help.

**Comment by gschueler** [comment]
Created a Trello card so you can vote for this feature: https://trello.com/c/knl0jfCj

**Comment by sean8sean** [comment]
+1 - we have shell and perl/python to handle a lot of tasks usually written as wrappers to an cmdline/exe, API, DB and many times we process against user supplied input files.  We want to give the users the power to run these.

**Comment by luisdavim** [comment]
+1 plese, this would be very useful.

**Comment by jasonschwab** [comment]
+1

Is this something that could possibly be implemented through a plugin?  

The scenario I envision is when someone goes to run the job a file upload dialog box is one of the options, and that file would be uploaded to the server and copied to a destination as part of the job, to be consumed by whatever the job is running (like a command like script).

I read over the manual around plugins as well as downloaded and looked at the code a bit and my first guess is that my scenario would not be able to be done via a plugin,  @gschueler could you confirm?

A plugin could probably be used to connect to Box or DropBox or something and get a file from there.

**Comment by gschueler** [comment]
@jasonschwab correct, there aren't currently any plugin points that would do this.  We have plans on the roadmap make the job Options pluggable, or to at least support File upload.

Please upvote this trello issue https://trello.com/c/knl0jfCj

**Comment by jasonschwab** [comment]
Thanks for the response, good to know.  I've voted.

**Comment by gregwjacobs** [comment]
+1

**Comment by lmarton** [comment]
+1 Same request here please - some folks need to upload files to be used during job runs. So upload file as an option and run the job.

**Comment by dsiluri** [comment]
+1

**Comment by dbh** [comment]
+1

**Comment by pviniciusfm** [comment]
+1

**Comment by pmsosa** [comment]
+1 This would be incredibly useful -- Any news on this? it seems like it's been on the back burner for a while.

**Comment by opm22** [comment]
+1!!!! We're also considering Rundeck to build a self-service interface for non-tech staff!

**Comment by ulrith** [comment]
+1

**Comment by atanass** [comment]
+1

**Comment by mabroor** [comment]
+1

**Comment by Miminoux** [comment]
+1

**Comment by WebSSH** [comment]
+1

**Comment by 0xAalaoui** [comment]
+1

**Comment by flacweb** [comment]
+1

**Comment by abeyt** [comment]
+1

**Comment by kino71** [comment]
+1

**Comment by rduarte-aig** [comment]
+1

**Comment by AndrewFarley** [comment]
+1 and +1'd on Trello

**Comment by schhabra-rtr** [comment]
+1

**Comment by meyju** [comment]
+1

**Comment by PetzJohannes** [comment]
+1

**Comment by genadipost** [comment]
+1

**Comment by thx8411** [comment]
+1

**Comment by ming871** [comment]
+1

**Comment by acqant** [comment]
+1 ( 2017 ! )

**Comment by ls457** [comment]
+1

**Comment by AndrewFarley** [comment]
+1 - So literally, what we do now to "avoid" this lack-of-feature is we take smaller binary-ish files (encrypted certificates) and base-64 them and paste them into a passworded text field.  This is quite honestly a ridiculous workaround.  I would personally sponsor a reasonable ($100-ish gift car) reward if we can get this feature added in the next release or two.  Not joking, it's just the way I can contribute to OSS because I literally have no time to implement this myself between running a business and doing client work.  Please implement this soon.

**Comment by gschueler** [comment]
@AndrewFarley that is a creative hack! I am working on this feature right now, due in an upcoming release. (PS, the best way to sponsor your favorite issue is as a Rundeck Pro subscriber: <http://rundeck.com>)

**Comment by AndrewFarley** [comment]
@gschueler I saw your merge request for this feature and I got really giddy.  Sorry, I can't afford Pro myself, nor can most of my clients, but I'll hold up my end of the bargain, I sponsor in my own way, but if I do run into a customer where I can get Rundeck Pro I will.  When that's merged upstream and available for use, ping me and choose a place (eg: amazon.com) for a gift card and its yours.  Cheers man, I'm super excited, thanks!
